### ISSUE TITLE
Questions for Richard

### ISSUE THREAD

**Issue by Myfanwy** [Myfanwy/gst_rheo#1]
Would it make sense to vary intercepts by fish and treatment?

Why would varying slopes make sense for these data?

**Comment by rmcelreath** [comment]
It's been long enough, that I'll need to refresh myself on the data structure.

My general rule: If the unit is repeated across cases, makes sense to put a varying effect on it. So multiple measures from same fish => varying effects on fish. Treatment is usually easier to think about as a regression effect, so estimate a slope for treatment, and varying that slope by fish could make sense, yes.

**Comment by Myfanwy** [comment]
Thanks Richard - I'm sorry, I was just taking notes for myself, didn't mean to flag you yet!  My question re: estimating a slope for treatment and then varying that slope by fish - that's what we do in the highest-weighted, non-centered model for the first experiment (m2NC).  My question is, does that mean the slopes of the treatments vary relative to each other as well as relative to each fish?  That is, are slopes for treatment A allowed to vary from fish to fish, then treatment B has a different slope relative to treatment A AND relative to each fish?

If so, I think that this model is probably over-fitted, since it's describing the data almost exactly, but not necessarily getting at the question at hand, which is for this population of green sturgeon (and hopefully, for another population of green sturgeon), how does treatment affect the proportion of time spent positively oriented?

In light of this, I was thinking of simplifying the model set, and having the most complex model include varying intercepts for fish and a fixed effect for treatment.  Alternatively, since the treatments are similar to each other pairwise (A is similar to B, and L is similar to D), I'd like to include a model that accounts for this similarity.  Something like:  

`logit(p) <- a + a_fish[fish_id] + (b_above*above + b_below*below)*a_velocity*velocity + (l_treatL*Light + d_treatD*Dark)*b_velocity*velocity`

Not sure if that's the correct code, though.

Thanks again for checking in!

**Comment by rmcelreath** [comment]
> My question is, does that mean the slopes of the treatments vary relative to each other as well as relative to each fish?

Yes. Both vary. This is the model in question, right?
```
    #likelihood
    y ~ dbeta2( p, theta ),
    
    # linear model
    logit(p) <- (b_above + ba_fish[fish_id])*above + (b_below + bb_fish[fish_id])*below + 
      (b_light + bl_fish[fish_id])*light + (b_dark + bd_fish[fish_id])*dark,
    
    # adaptive NON-CENTERED priors 
    c(ba_fish, bb_fish, bl_fish, bd_fish)[fish_id] ~ dmvnormNC(sigma_fish, Rho_fish),
    # fixed priors
    c(b_above, b_below, b_light, b_dark) ~ dnorm(0,1),
    theta ~ dcauchy(0,1),
    sigma_fish ~ dexp(1),
    Rho_fish ~ dlkjcorr(2)
```
That looks okay to me. One way to think of this is that the model estimates an average behavior for each treatment (above, below, light, dark), as well as accounting for variation among fish (and correlations across treatments by fish).

(Also, we are saying "slopes" here, but really these are just unique intercepts for each treatment, with deviations from the average by fish.)

(Are there replicates of each treatment for each fish? If not, doubt there's much information in the data about the ba_fish, bb_fish, bl_fish, bd_fish parameters. An intercept for each fish might be just as good.)

Anyway, the b_above, b_below, b_light, b_dark parameters provides a partial answer to your question: how does treatment affect the proportion of time spent positively oriented? And then the variation across fish provides a more complete answer, because even if the average fish doesn't respond to a treatment, it may be that many fish do. Or some fish might vary overall in positive orientation, so accounting for that variation will provide a better estimate of the average.

Anyway, since the fish effects are pooled, I wouldn't worry about overfitting. Doesn't sound like WAIC is worried about overfitting either.

Your velocity model makes sense to me:
```
logit(p) <- a + a_fish[fish_id] + (b_above*above + b_below*below)*a_velocity*velocity + (l_treatL*Light + d_treatD*Dark)*b_velocity*velocity
```
This says that fish vary overall in positive orientation (a_fish), and then everything else is constant across fish.

**Comment by Myfanwy** [comment]
> (Also, we are saying "slopes" here, but really these are just unique intercepts for each treatment, with deviations from the average by fish.)

Ok I think this was tripping me up.  Because Treatment is categorical, I wasn't understanding how slopes could vary across each treatment (there's no gradient in the Treatments that a fish could respond to, so I wasn't conceptualizing the varying slopes very well).  But you're saying that the "varying slopes" in model m2NC are actually unique intercepts - so the fish has it's own unique intercept (estimated from `ba_fish[fish_id], etc`, and then the estimates for treatment (`ba_above, etc`) correspond to its own coefficient for that treatment, for that fish?  Sorry to belabor this, but just to make sure I understand, is this sketch a correct representation of the difference between varying (random) intercepts & fixed treatment coefficients (slopes) and varying fish intercepts and treatment coefficients?  Just like the chimpanzee example, but applied to this data?

![modelvizs](https://cloud.githubusercontent.com/assets/6519519/17939883/5011d482-69e1-11e6-9ae7-7795abdf1ff4.jpg)


As for the m2NC model, for the first experiment we don't really have repeat measures on individuals; each fish underwent each treatment once, (except for a single fish that was run through each treatment twice... sigh).  In the second experiment, some of the fish were used again for the L and D treatments, with some new fish thrown in.  So it might make sense to pull out the bb_fish[fish_id] into a simple a_fish[fish_id] estimate for the first experiment, but not the second - I think you had me format the model this way because it made the treatment coefficients more interpretable to me at the time, but if I switched it back, (`logit(p) ~ a_fish[fish_id] + a_above*Above + b_below*Below, etc,` with adaptive priors on the treatment parameters to allow for varying coefficients on treatment, then the treatment coefficients would just be the deviations from the mean estimate of proportion of time spent positive for each fish, correct?

Sorry - I know we went through all this when you were here in Davis.  I took a long break from it because I had handed it off to the original researcher, but he's thrown it back to me, so I'm revisiting it all and making it a chapter in my dissertation (having spent so much time on it I figured I should turn it into an actual investment).

Thanks again, hope you're well!

**Comment by rmcelreath** [comment]
Yeah, I think you get it. In your first sketch, fish vary only in overall orientation. In your second sketch, fish vary all over, despite any average difference by treatment.

It may very well be that the first sketch is a sufficient model of the sample. I suspect fitting the second model would show whether that is the case.

For interpreting parameters, it'll depend upon the exact way you structure the linear model, whether the mean effects are in the priors, etc. If you do something like:
```
logit(p) <- a + a_fish[fish_id] + a_treatment[treatment_id]
```
Then all the ``a_fish`` and ``a_treatment`` parameters are deviations from the overall mean ``a`` across fish and treatments.
